From f1a8c91d55ea26db1af273634afa0bf9304af6f8 Mon Sep 17 00:00:00 2001
From: Michael Krummsdorf <michael.krummsdorf@tqs.de>
Date: Wed, 28 May 2014 14:42:02 +0200
Subject: [PATCH] tqma28x: improve qualified pin setup

first patch to rely on qualified pin setup
was not perfectly correct.

Signed-off-by: Michael Krummsdorf <michael.krummsdorf@tqs.de>
---
 arch/arm/boot/dts/imx28-mba28.dts     |   10 ++++------
 arch/arm/boot/dts/imx28-mba28l-ab.dts |   11 +++++++----
 arch/arm/boot/dts/imx28-tqma28.dtsi   |   10 ++++++++++
 arch/arm/boot/dts/imx28-tqma28l.dtsi  |    6 +++++-
 4 files changed, 26 insertions(+), 11 deletions(-)

diff --git a/arch/arm/boot/dts/imx28-mba28.dts b/arch/arm/boot/dts/imx28-mba28.dts
index e2f94eb..2f66c95 100644
--- a/arch/arm/boot/dts/imx28-mba28.dts
+++ b/arch/arm/boot/dts/imx28-mba28.dts
@@ -89,7 +89,7 @@
 					>;
 					fsl,drive-strength = <1>;
 					fsl,voltage = <1>;
-					fsl,pull-up = <1>;
+					fsl,pull-up = <0>;
 				};
 
 				mmc1_cd_cfg: mmc1-cd-cfg {
@@ -103,16 +103,14 @@
 					fsl,pinmux-ids = <
 						0x0191 /* MX28_PAD_GPMI_WRN__SSP1_SCK */
 					>;
+					fsl,voltage = <1>;
 					fsl,drive-strength = <1>;
-					fsl,pull-up = <0>;
+					fsl,pull-up = <1>;
 				};
 
 				pwm2_pins_a: pwm2@0 {
 					fsl,drive-strength = <1>;
-				};
-
-				duart_pins_a: duart@0 {
-					fsl,drive-strength = <1>;
+					fsl,pull-up = <1>;
 				};
 
 				auart0_pins_a: auart0@0 {
diff --git a/arch/arm/boot/dts/imx28-mba28l-ab.dts b/arch/arm/boot/dts/imx28-mba28l-ab.dts
index 62d1359..896a185 100644
--- a/arch/arm/boot/dts/imx28-mba28l-ab.dts
+++ b/arch/arm/boot/dts/imx28-mba28l-ab.dts
@@ -77,8 +77,14 @@
 					fsl,drive-strength = <1>;
 				};
 
+				mmc0_4bit_pins_a: mmc0-4bit@0 {
+					fsl,pull-up = <0>;
+				};
+
 				mmc0_sck_cfg: mmc0-sck-cfg {
+					fsl,voltage = <1>;
 					fsl,drive-strength = <1>;
+					fsl,pull-up = <1>;
 				};
 
 				i2c0_pins_a: i2c0@0 {
@@ -87,10 +93,7 @@
 
 				pwm4_pins_a: pwm4@0 {
 					fsl,drive-strength = <1>;
-				};
-
-				duart_pins_a: duart@0 {
-					fsl,drive-strength = <1>;
+					fsl,pull-up = <1>;
 				};
 
 				auart0_pins_a: auart0@0 {
diff --git a/arch/arm/boot/dts/imx28-tqma28.dtsi b/arch/arm/boot/dts/imx28-tqma28.dtsi
index 324c58a..30b82f5 100644
--- a/arch/arm/boot/dts/imx28-tqma28.dtsi
+++ b/arch/arm/boot/dts/imx28-tqma28.dtsi
@@ -28,6 +28,16 @@
 				vmmc-supply = <&reg_vddio_sd0>;
 				status = "okay";
 			};
+
+			pinctrl@80018000 {
+				mmc0_8bit_pins_a: mmc0-8bit@0 {
+					fsl,drive-strength = <0>;
+				};
+
+				mmc0_sck_cfg: mmc0-sck-cfg {
+					fsl,voltage = <1>;
+				};
+			};
 		};
 
 		apbx@80040000 {
diff --git a/arch/arm/boot/dts/imx28-tqma28l.dtsi b/arch/arm/boot/dts/imx28-tqma28l.dtsi
index 4a180c3..601ad7e 100644
--- a/arch/arm/boot/dts/imx28-tqma28l.dtsi
+++ b/arch/arm/boot/dts/imx28-tqma28l.dtsi
@@ -44,10 +44,14 @@
 						0x0071 /* MX28_PAD_GPMI_D07__SSP1_D7 */
 						0x0151 /* MX28_PAD_GPMI_RDY1__SSP1_CMD */
 					>;
-					fsl,drive-strength = <1>;
+					fsl,drive-strength = <0>;
 					fsl,voltage = <1>;
 					fsl,pull-up = <1>;
 				};
+
+				mmc1_sck_cfg: mmc1-sck-cfg {
+					fsl,voltage = <1>;
+				};
 			};
 		};
 	};
-- 
1.7.10.4

